gammapy/irf/core.py

```diff
@@ -201,8 +202,20 @@ def evaluate(self, method=None, **kwargs):
             coord = kwargs.get(key, value)
             if coord is not None:
                 coords_default[key] = u.Quantity(coord, copy=False)
+        data = self._interpolate(coords_default.values(), method=method)
 
-        return self._interpolate(coords_default.values(), method=method)
+        if self.fill_value != self.default_interp_kwargs["fill_value"]:
```

Comment:
Do we need the extra `.fill_value` definition here? Why can't we just rely on `default_interp_kwargs["fill_value"]`?

Reply:
yes as default_interp_kwargs["fill_value"] must remain None so the extrapolation is performed.

Reply:
I see, maybe we just handle it in `_interpolate` instead , like:
```python3
kwargs = self.default_interpolation_kwargs.copy()
# Allow extrap[olation with in bins
kwargs["fill_value"] = None
return ScaledRegularGridInterpolator(..., **kwargs)
```
And later rely on:
```python3
if self.default_interpolation_kwargs["fill_value"] is not None:
    # replace values outside by fill value
```
The fill values are then just defined on a sub-class level...

Reply:
This means we cannot disable completely the extrapolation within bins if necessary but ok.

Reply:
I think that's fine for now. It's the default behaviour we want to establish...